app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventRuleEngineRepository.java

```diff
@@ -0,0 +1,87 @@
+package org.dhis2.usescases.eventsWithoutRegistration.eventCapture;
+
+import androidx.annotation.NonNull;
+
+import org.dhis2.Bindings.RuleExtensionsKt;
+import org.dhis2.data.forms.FormRepository;
+import org.dhis2.data.forms.dataentry.RuleEngineRepository;
+import org.dhis2.utils.Result;
+import org.hisp.dhis.android.core.D2;
+import org.hisp.dhis.android.core.event.Event;
+import org.hisp.dhis.android.core.organisationunit.OrganisationUnit;
+import org.hisp.dhis.android.core.program.ProgramStage;
+import org.hisp.dhis.rules.RuleEngine;
+import org.hisp.dhis.rules.models.RuleDataValue;
+import org.hisp.dhis.rules.models.RuleEffect;
+import org.hisp.dhis.rules.models.RuleEvent;
+
+import java.util.List;
+
+import io.reactivex.Flowable;
+
+public final class EventRuleEngineRepository implements RuleEngineRepository {
```

Comment:
Do you think that we could add tests for this class?

Reply:
done! :)